website/docs/commands/publishing/publish.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Publish
+sidebar_position: 20
+---
+
+import {ChainedSnippets} from "../../../src/components/MarkdownComponents.js";
+
+The `publish` sub-command allows to publish Scala CLI projects to Maven repositories.
+
+## Required settings
+
+`scala-cli publish` and `scala-cli publish local` might complain about missing settings.
+An organization, a name (or a module name), and (a way to compute) a version are needed, but Scala CLI may
+be able to compute sensible defaults for them.
+
+We recommend setting the settings below via using directives rather than on the command-line,
+so that commands such as `scala publish .` or `scala publish local .` work fine for your
+project. Command-line options for those settings take over the using directive values, and are
+provided as a convenience.
+
+This table lists settings allowing to specify those. See the sub-sections right after for more details.
+
+|          | `using` directive | Command-line option | Example values | Notes |
+|----------|-------------------|---------------------|----------------|-------|
+| Organization | `publish.organization` | `--organization` | `org.virtuslab.scala-cli` | |
+| Name | `publish.name` | `--name` | `scala-cli` | |
+| Module Name | `publish.moduleName` | `--module-name` | `scala-cli_3` | Module Name includes the Scala prefix, such as `_2.13` or `_3`. Specifying Name should be favored over Module Name |
+| Compute Version | `publish.computeVersion` | `--compute-version` | `git:tag` | |
+| Version | `publish.version` | `--version` | `0.1.0`, `0.1.1-SNAPSHOT` | As much as possible, Compute Version (describing how to compute the version) should be favored over Version |
+
+### Organization
+
+If your Scala CLI project lives in a git repository having a GitHub remote, Scala CLI
+will infer an organization from it: if your project lives in GitHub organization `foo`
+(that is, lives somewhere under `https://github.com/foo/`), Scala CLI will use
+`io.github.foo` as default Maven organization.
+
+To override this default value, set the `publish.organization` directive, like
+```scala
+//> using publish.organization "io.github.foo"
+```
+
+### Name
+
+Scala CLI will use the project directory name as default Maven name. That is, if your
+Scala CLI project lives in a directory named `something`, it will be published as
+`something` (pure Java project) or `something_3` (Scala 3 project) for example.
+
+To override this default value, set the `publish.name` directive, like
+```scala
+//> using publish.name "something"
+```
+
+### Version
+
+If your Scala CLI project lives in a git repository, Scala CLI will infer a way to compute
+versions from it: if the current commit has a tag `v1.2.3`, version `1.2.3` is assumed.
+Else, if it has such a tag earlier in the git history, version `1.2.4-SNAPSHOT` is assumed.
+
+To override this default value, set the `publish.computeVersion` directive, like
+```scala
+//> using publish.computeVersion "git:tag"
```

Comment:
I think we should link here to a doc explaining what `computeVersions` are supported and how each work